tests/cocotb/basic/test/test_hwpe_stream_split.py

```diff
@@ -0,0 +1,301 @@
+
+#-----------------------------------
+# Add testbench to RTL list
+#-----------------------------------
+tb_path = basic_path + '/wrappers/wrapper_hwpe_stream_split.sv'
+rtl_sources.append(tb_path)
+
+#-----------------------------------
+# Verification functions
+#-----------------------------------
+
+#-----------------------------------
+# Main test bench
+#-----------------------------------
+# For the main test bench, we need to make sure the ports
+# are consistent with the DUT. Double check the main module.
+#-----------------------------------
+@cocotb.test()
+async def hwpe_stream_split(dut):
+
+    #-----------------------------------
+    # Local parameters
+    # Don't touch these
+    #-----------------------------------
+    STRB_WIDTH_IN  = int(DATA_WIDTH_IN / 8)  
+    MAX_DATA_VAL_I = 2**(DATA_WIDTH_IN)-1      
+    MAX_STRB_VAL_I = 2**(STRB_WIDTH_IN)-1
+    DATA_WIDTH_OUT = int(DATA_WIDTH_IN/NB_OUT_STREAMS)
+    STRB_WIDTH_OUT = int(DATA_WIDTH_OUT/8)
+
+    # Check first DATA_WIDTH_IN is multiple of 8
+    assert ((DATA_WIDTH_IN % 8) == 0), f"{DATA_WIDTH_IN} is not a multiple of 8!"
+
+    #-----------------------------------
+    # TB parameters:
+    # NB_OUT_STREAMS - indicates how many input streams
+    # DATA_WIDTH_IN    - indicates literal data width
+    # TB drivers ports:
+    # logic clk_i
+    # logic rst_ni
+    # logic clear_i
+    # hwpe_stream_intf_stream.sink push_i [NB_OUT_STREAMS-1:0],
+    # >> input  valid_i
+    # >> input   data_i [DATA_WIDTH_IN-1:0]
+    # >> input   strb_i [STRB_WIDTH_IN-1:0]
+    # >> output ready_i
+    # hwpe_stream_intf_stream.source pop_o
+    # >> output valid_o
+    # >> output  data_o [DATA_WIDTH_OUT-1:0]
+    # >> output  strb_o [STRB_WIDTH_OUT-1:0]
+    # >> input  ready_o
+    #-----------------------------------
+
+    # Initialize clock
+    clock = Clock(dut.clk_i, 10, units="ns")
+    cocotb.start_soon(clock.start())
+
+    # Reset or intial values
+    dut.rst_ni.value  = 0
+    dut.clear_i.value = 0
+
+    # Initialize push_i input values
+    dut.valid_i.value = 0
+    dut.data_i.value  = 0
+    dut.strb_i.value  = 0
+    
+    # Initialize pop_o values
+    for i in range(NB_OUT_STREAMS):
+        dut.ready_o[i].value = 0
+
+    # Wait 2 cycles to reset
+    await RisingEdge(dut.clk_i)
+    await RisingEdge(dut.clk_i)
+
+    # Deassert reset
+    dut.rst_ni.value = 1
+
+    #-----------------------------------
+    # Testing check
+    #-----------------------------------
+
+    cocotb.log.info(f'------------------------------------ START OF TESTING ------------------------------------')
+    cocotb.log.info(f'Running parameters:')
+    cocotb.log.info(f'NB_OUT_STREAMS:{NB_OUT_STREAMS}')
+    cocotb.log.info(f'DATA_WIDTH_IN :{DATA_WIDTH_IN}')
+    cocotb.log.info(f'RANDOM_SEED   :{RANDOM_SEED}')
+    cocotb.log.info(f'------------------------------------------------------------------------------------------')
```

Comment:
This seems also copy-pasted from the other testbench and could be written more concisely in my opinion

Reply:
OK!